kong/router/atc.lua

```diff
@@ -407,12 +408,14 @@ function _M:matching(params)
                       nil, nil,
                       params.sni, params.headers, params.queries)
 
+  self.context:reset()
```

Comment:
We should move this to just before `fill_atc_context`, to stay consistent with stream and make it more clear that the context was reset before used.

Reply:
done